src/form.js

```diff
@@ -320,7 +320,11 @@ define([
                         meta.attributes.src = this.knownInstances[attrs.id];
                     }
                 } else if (this.knownInstances.hasOwnProperty(attrs.id)) {
-                    attrs.src = this.knownInstances[attrs.id];
+                    if (_.isObject(this.knownInstances[attrs.id]))
```

Comment:
This could accidentally catch a string created with `new` (not that we ever do that, but...). It would be a little safer to do it this way:
```js
if (_.isString(this.knownInstances[attrs.id])) {
    attrs.src = this.knownInstances[attrs.id];
} else {
    attrs.children = this.knownInstances[attrs.id];
}
```
Although even that has a smell. What about making the values in `knownInstances` be an object like `{src: <src uri>}` or `{children: <children obj>}`? Then you could iterate over properties of the value and assign all found attributes to `attrs`. The bonus of this approach is that it does not require a type check and we could easily add more attributes in the future if needed.

Also, the if body should be delimited with `{}`.